galsim/scene.py

```diff
@@ -350,13 +504,24 @@ def makeGalaxy(self, index=None, gal_type=None, chromatic=False, noise_pad_size=
                 raise RuntimeError("Cannot yet make real chromatic galaxies!")
             gal_list = self._makeReal(indices, noise_pad_size, rng, gsparams)
         else:
+            # If no pre-selection was done based on radius or flux, then we won't have checked
+            # whether we're using the old or new catalog (the latter of which has a lot of
+            # precomputations done).  Just in case, let's check here, though it does seem like a bit
+            # of overkill to emit this warning each time.
+            if 'hlr' not in self.param_cat.dtype.names:
+                import warnings
+                warnings.warn(
+                    'You seem to have an old version of the COSMOS parameter file.\n'+
+                    'Please run `galsim_download_cosmos` to re-download the COSMOS catalog\n' +
+                    'and take advantage of pre-computation of many quantities..')
+
             gal_list = self._makeParametric(indices, chromatic, sersic_prec, gsparams)
 
-        # If deep, rescale the size and flux
+        # If trying to use the 23.5 sample and "fake" a deep sample, rescale the size and flux as
+        # suggested in the GREAT3 handbook.
         if deep:
```

Comment:
Should the `deep` kwarg be disabled if using the 25.2 catalog?  (Same question for `_makeSingleGalaxy` down below).